ospfd/ospf_vty.c

```diff
@@ -1015,80 +1016,44 @@ DEFUN (ospf_area_vlink,
 		return ospf_vl_set(ospf, &vl_config);
 	}
 
-	/* Deal with other parameters */
-	for (i = 5; i < argc; i++) {
-
-		/* vty_out (vty, "argv[%d]->arg - %s\n", i, argv[i]->text); */
-
-		switch (argv[i]->arg[0]) {
-
-		case 'a':
-			if (i > 5
-			    || strncmp(argv[i]->arg, "authentication-", 15)
-				       == 0) {
-				/* authentication-key - this option can occur
-				   anywhere on
-							command line.  At start
-				   of command line
-							must check for
-				   authentication option. */
-				memset(auth_key, 0, OSPF_AUTH_SIMPLE_SIZE + 1);
-				strncpy(auth_key, argv[i + 1]->text,
-					OSPF_AUTH_SIMPLE_SIZE);
-				vl_config.auth_key = auth_key;
-				i++;
-			} else if (strncmp(argv[i]->arg, "authentication", 14)
-				   == 0) {
-				/* authentication  - this option can only occur
-				   at start
-						     of command line */
-				vl_config.auth_type = OSPF_AUTH_SIMPLE;
-				if ((i + 1) < argc) {
-					if (strncmp(argv[i + 1]->arg, "n", 1)
-					    == 0) {
-						/* "authentication null" */
-						vl_config.auth_type =
-							OSPF_AUTH_NULL;
-						i++;
-					} else if (
-						strncmp(argv[i + 1]->arg, "m",
-							1)
-							== 0
-						&& !strmatch(
-							   argv[i + 1]->text,
-							   "message-digest-")) {
-						/* "authentication
-						 * message-digest" */
-						vl_config.auth_type =
-							OSPF_AUTH_CRYPTOGRAPHIC;
-						i++;
-					}
-				}
-			}
-			break;
+	if (argv_find(argv, argc, "authentication", &idx)) {
+		/* authentication  - this option can only occur
+		at start of command line */
+		vl_config.auth_type = OSPF_AUTH_SIMPLE;
+	}
 
-		case 'm':
-			/* message-digest-key */
-			i++;
-			if (i < argc) {
-				vl_config.crypto_key_id =
-					strtol(argv[i]->arg, NULL, 10);
-				if (vl_config.crypto_key_id < 0)
-					return CMD_WARNING_CONFIG_FAILED;
-				i++;
-				if (i < argc) {
-					memset(md5_key, 0,
-					       OSPF_AUTH_MD5_SIZE + 1);
-					strncpy(md5_key, argv[i]->arg,
-						OSPF_AUTH_MD5_SIZE);
-					vl_config.md5_key = md5_key;
-				}
-			} else
-				vl_config.md5_key = NULL;
-			break;
-		}
+	idx = 2;
```

Comment:
This shouldn't be necessary, `argv_find()` takes care of updating `idx` for us.

In general, once `idx` is initialized to zero, you don't need to care about it anymore. The only exception I'm aware of is when you have parameters inside curly braces. @qlyoung please correct me if I'm wrong :)